user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -847,6 +847,13 @@ define([
       }
       //require folders header template
       $folderBrowserContainer.prepend($(templates.folderBrowserHeader(obj)));
+      $folderBrowserContainer.find("#refreshBrowserIcon").bind("keydown", function (event) {
+        let keyCode = event.which || event.keyCode;
+        // ENTER , SPACE
+        if (keyCode === 13 || keyCode === 32) {
+          obj.refreshHandler();
+        }
+      })
```

Comment:
Please, use the ARIA button pattern as implemented here: https://www.w3.org/WAI/ARIA/apg/example-index/button/js/button.js.

Reply:
I've shared this adaptation I wrote with Vamsi:

```js
// Based on https://www.w3.org/WAI/ARIA/apg/example-index/button/js/button.js
function makeAccessibleActionButton(elem, options) {
  var handler = options.handler;
  var label = options.label;

  elem.role = "button";
  elem.tabIndex = 0;
  elem.addEventListener('click', handler);
  elem.addEventListener('keydown', actionButtonKeydownHandler);
  elem.addEventListener('keyup', actionButtonKeyupHandler);

  var KEYCODE_ENTER = 13;
  var KEYCODE_SPACE = 32;

  function actionButtonKeydownHandler(event) {
    // The action button is activated by space on the keyup event, but the
    // default action for space is already triggered on keydown. It needs to be
    // prevented to stop scrolling the page before activating the button.
    if (event.keyCode === KEYCODE_SPACE) {
      event.preventDefault();
    } else if (event.keyCode === KEYCODE_ENTER) {
      // If enter is pressed, activate the button
      event.preventDefault();
      handler(event);
    }
  }

  function actionButtonKeyupHandler(event) {
    if (event.keyCode === KEYCODE_SPACE) {
      event.preventDefault();
      handler(event);
    }
  }
}

// Based on https://www.w3.org/WAI/ARIA/apg/example-index/button/js/button.js
function makeAccessibleToggleButton(elem, options) {
  var handler = options.handler;

  var options2 = Object.create(options);
  options2.handler = function(event) {
    toggleButtonState();
    handler(event);
  };

  var ICON_MUTE_URL = '#icon-mute';
  var ICON_SOUND_URL = '#icon-sound';

  makeAccessibleActionButton(elem, options2);

  function toggleButtonState() {
    var isAriaPressed = elem.getAttribute("aria-pressed") === "true";
  
    elem.setAttribute("aria-pressed", String(!isAriaPressed));
  
    var icon = elem.querySelector("use");
    if(icon) {
      icon.setAttribute("xlink:href", isAriaPressed ? ICON_SOUND_URL : ICON_MUTE_URL);
    }
  }
}
```

Given we appear to be needing these in multiple places (mantle and plugins), we should create a Common-UI module to hold this kind of utilities (e.g. in an AMD module such as `common-ui/util/_a11y`).

Reply:
done. will be creating a PR.